chia/wallet/puzzles/dao_treasury.clsp

```diff
@@ -85,6 +98,7 @@
                     PASS_MARGIN
                     proposal_announcement
                     proposal_validator_solution
+                    ; [REVIEW - architecture] does the proposal validator really need the conditions here?  I would assume it's checking the proposal's puzzle which means that it knows the likely set of conditions it could possibly output.
```

Comment:
We might want validators that prevent certain conditions or certain types of announcements

Reply:
Hmm yeah, I was thinking that the proposal could just check that, but I guess the proposal doesn't normally run the delegated puzzle it's announcing.

Reply:
Well actually there's still some argument to omit this.  The security of the delegated puzzle is that it gets voted on.  If there are certain conditions that are banned, that could be a layer in the dpuz, and the proposal could check that the dpuz is of that format.  All you'd have to change in the treasury for this is the PROPOSAL_MOD instead of the whole PROPOSAL_VALIDATOR which is potentially more dangerous.  Additionally, it optimizes most of the other spend cases where a dpuz is allowed to output whatever it wants.  I don't feel super strongly about this though, I just think it would be a little bit better.